### DISCUSSION TITLE
Incorrect value of smem output for processes running in firejail

### DISCUSSION THREAD

**Discussion by Rosika2** [netblue30/firejail#5193] [Q&A] [answered]
Hi all,

I wonder if anyone could help me with a curious problem.

I´ve just installed `smem` ("Report memory usage with shared memory divided proportionally") for occasionally taking a closer look at the RAM.

- https://manpages.ubuntu.com/manpages/jammy/man8/smem.8.html
- https://www.howtogeek.com/773161/easily-understand-your-linux-ram-usage-with-smem/  # pretty good introduction to the topic

Example: # I´ve got 4 GB of RAM, `firefox` is running within `firejail`

`smem -c pss -P firefox -k -t | tail -n 1  # quick way to find out the RAM consumption of a certain program and its child processes`

yields:

`7.2M`

...which seems far too low bearing in mind `conky` says firefox uses 12 percent of RAM (4 GB in total). So this should be arount 480 MB.

Out of curiosity I shut down firefox and started a non-sandboxed instance of it. NOW things look more realistic:

```
smem -c pss -P firefox -k -t | tail -n 1
  463.8M
```

So it seems any process running within firejail is just displayed with a fraction of its actual RAM usage.

When running the `smem` command as **root** however, the values seem to make more sense again. Here firefox is running within firejail:

```
sudo smem -c pss -P firefox -k -t | tail -n 1
    818.0M
```

My question is: It looks like processes running within firejail are not accounted for by smem if smem is not run with "sudo". But why might that be?
I was of the opinion that e.g. "firefox" running within firejail was still a process issued by **user** _rosika_ (in my case).

`ps aux | grep firefox`

would suggest that, I guess:

`rosika 37530 0.0 0.0 5336 2504 ? S 16:28 0:00 firejail --private=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/work2 firefox`

Many thanks for your help in advance and many greetings.
Rosika

P.S.:

my system: Linux/Lubuntu 20.04.4 LTS, 64 bit

**Comment by rusty-snake** [comment, marked as answer]
Various files in `/proc/<pid>` can not be accessed if the user-ns in which `<pid>` resides is owned by the root user in the current user-ns (and you are not root). edit: smem relays on `/proc/<pid>/smaps`. 

Less technical: smem (as user) does not work with firejails created with `noroot`-command.

**Comment by Rosika2** [comment]
Hi @rusty-snake, :wave:

thanks so much for your reply and excellent explanation of how things work under the hood.

I guess with "user-ns" you mean "user namespace".
I tried to re-create the paths you were referring to and indeed everything´s llike you said:

The command `firejail --private=[...] firefox` has got the PID 2285; so I looked it up:

```
rosika@rosika-10159 ~ [1]> ll /proc | grep 2285
dr-xr-xr-x  9 rosika           rosika              0 Jun 16 12:37 2285
rosika@rosika-10159 ~> ll /proc/2285/smaps
-r--r--r-- 1 root root 0 Jun 16 12:57 /proc/2285/smaps
```

> Less technical: smem (as user) does not work with firejails created with noroot-command.

I also could verify that in `/etc/firejail/firefox.profile` there is the following entry:
`#ignore noroot`. 
So everything makes sense now.

In order to sum up everything running in RAM `smem` has to be run as root then.
Good to know. :wink:

Thanks again for taking the time to explain everything in such detail. It´s highly appreciated.

Many greetings from Rosika. :smile:
